### PR TITLE
Speedup version bump tests

### PR DESCRIPTION
**Pull request by jhrcek**
### Description of the change

Attempt to speed up test suite by focusing on the slowest tests as described in https://github.com/spacchetti/spago/issues/440

Locally this change seems to have shaved off ~20s of test running time.

### Checklist:

- [x] Added the change to the "Unreleased" section of the changelog
- [ ] Added some example of the new feature to the `README`
- [x] Added a test for the contribution (if applicable)

### CODE REVIEW

File: src/Spago/Version.hs [hunk 1]
```diff
@@ -55,18 +67,18 @@ getCurrentVersion = do
 
 
 -- | Get the next version to use, or die if this would result in the version number going down/not changing.
-getNextVersion :: Spago m => VersionBump -> SemVer -> m SemVer
+getNextVersion :: VersionBump -> SemVer -> Either Text SemVer
```
Comment (jhrcek) [inline src/Spago/Version.hs]: Made this pure so it's easier to unit test..

File: test/BumpVersionSpec.hs [hunk 2]
```diff
@@ -53,70 +59,118 @@ setOverrides overrides = do
...
-    before_ (initGitTag "v1.3.4") $ it "Spago should bump patch version" $ do
+      it "Spago should complain when no git repo exists" $ do
 
-      spago ["bump-version", "--no-dry-run", "patch"] >>= shouldBeSuccess
-      getHighestTag >>= (`shouldBe` Just "v1.3.5")
+        spago ["bump-version", "minor"] >>= shouldBeFailureInfix
+          "Your git working tree is dirty. Please commit or stash your changes first"
 
-    before_ (initGitTag "v1.3.4") $ it "Spago should bump minor version" $ do
+      before_ (initGitTag "v1.2.3") $ it "Spago should only make a tag with `--no-dry-run`" $ do
```
Comment (jhrcek) [inline test/BumpVersionSpec.hs]: The basic idea was to replace [these tests](https://github.com/spacchetti/spago/blob/52ddd50747bd139474008780cc612396a4b5477b/test/BumpVersionSpec.hs#L73-L94) with just one - they all test the same path, just differ in git tag version, which I think is adequately tested in the unit/property tests above.

File: src/Spago/Version.hs (hunk 1 above)
Comment (Dretch) [reply src/Spago/Version.hs]: Seems reasonable. It is possibly worth re-wording the comment so it doesn't mention `die`.

File: test/BumpVersionSpec.hs [hunk 3]
```diff
@@ -53,70 +59,118 @@ setOverrides overrides = do
...
+        parseVersionBump "1.2.3" `shouldBe` Just (Exact (SemVer 1 2 3 [] []))
 
-    before_ (initGitTag "v1.2.3") $ it "Spago should not make a tag when not passing --no-dry-run" $ do
+      -- TODO is this desired behavior, or should we just drop ONE 'v'? I'd agree it's edge case, but still :-)
+      it "should drop multiple 'v's from the beginning" $
+        parseVersionBump "vvvvvvvv1.2.3" `shouldBe` Just (Exact (SemVer 1 2 3 [] []))
 
-      spago ["bump-version", "minor"] >>= shouldBeSuccess
-      getHighestTag >>= (`shouldBe` Just "v1.2.3")
+      -- TODO should this work or should we strip these in parser implementation?
```
Comment (Dretch) [inline test/BumpVersionSpec.hs]: PREREL / META should probably be allowed - for one thing because there are existing packages that have such versions - e.g. `halogen` `v5.0.0-rc.7`.

File: test/BumpVersionSpec.hs [hunk 4]
```diff
@@ -53,70 +59,118 @@ setOverrides overrides = do
...
+      it "should parse version starting with 'v'" $
+        parseVersionBump "v1.2.3" `shouldBe` Just (Exact (SemVer 1 2 3 [] []))
 
-      spago ["bump-version", "minor"] >>= shouldBeFailureInfix
-        "Your git working tree is dirty. Please commit or stash your changes first"
+      it "should parse version not starting with 'v'" $
+        parseVersionBump "1.2.3" `shouldBe` Just (Exact (SemVer 1 2 3 [] []))
 
-    before_ (initGitTag "v1.2.3") $ it "Spago should not make a tag when not passing --no-dry-run" $ do
+      -- TODO is this desired behavior, or should we just drop ONE 'v'? I'd agree it's edge case, but still :-)
```
Comment (Dretch) [inline test/BumpVersionSpec.hs]: I don't know. :man_shrugging:

File: test/BumpVersionSpec.hs (hunk 2 above)
Comment (Dretch) [reply test/BumpVersionSpec.hs]: :+1:

File: src/Spago/Version.hs (hunk 1 above)
Comment (jhrcek) [reply src/Spago/Version.hs]: Good point. I actually went for deleting this comment, as I find the code self-explanatory enough.

**Comment by jhrcek** [comment]
@f-f just wanted to check with you what's the general strategy around PR reviews. Should I wait for your review/approval before I merge, or is just one review enough, like in the case of this PR?

**Comment by f-f** [comment]
@jhrcek great point, thanks for asking. This has not come up before so that's why we don't have a written policy about it yet

I think a review from a contributor is enough to merge, no need to wait for my review :)

**Comment by f-f** [comment]
@jhrcek thanks for doing this 🙂
